src/oeis_sequence.cpp

```diff
@@ -278,14 +275,22 @@ const Sequence& OeisSequence::getFull() const
       }
     }
   }
-  return full;
+  real_max_terms = std::min( real_max_terms, full.size() );
+  if ( real_max_terms == full.size() )
+  {
+    return full;
+  }
+  else
+  {
+    return Sequence( std::vector<number_t>( full.begin(), full.begin() + real_max_terms ) );
+  }
 }
 
 void OeisSequence::fetchBFile() const
 {
   if ( !attempted_bfile )
   {
-    getFull();
+    getTerms( -1 );
```

Comment:
I wonder if it would make sense for clarity to keep `getFull()` (can internally call `getTerms(-1)`)